src/steps/iNavigateTo.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * This step will set a new hash to the app.
+ * The app should react to the navigation by providing proper routing.
+ *
+ * ToDo: Add [with <JSON>] to synopsis to pass data as json
+ *    regex: "(\\s+with\\s+(.+?))?"
+ *    parsing: var oObject = JSON.parse(sObject);
+ */
+module.exports = {
+    docs: {
+        description: "Performs navigation using hash.",
+        synopsis: "I navigate to <hash>",
+        examples: [
+            "I navigate to /Customers/4711",
+            "I navigate to /Customers/4711?&tab=person"
+        ]
+    },
+    regexp: new RegExp([
+        "^",                          //start of string
+        "I\\snavigate\\sto\\s",       //enty point
+        "([=a-zA-Z0-9\/\?\&]+){1}",   //hash with parameters
```

Comment:
Would use a more expressive regex here: `"(/|((/[a-zA-Z0-9_]+)+([?]([&a-zA-Z0-9_]+?=[^&?]*?)+)?))"` . You don't need to escape all characters when using a character class see https://www.regular-expressions.info/refcharclass.html .